src/directives/layertree.js

```diff
@@ -439,6 +439,60 @@ ngeo.LayertreeController.prototype.setDataSource = function(dataSource) {
 
 
 /**
+ * Determines whether the layer tree controller supports being customized.
+ * For example, having its layer opacity changed, displaying its legend, etc.
+ *
+ * If any requirement is met, then the treeCtrl is considered supporting
+ * "customization", regardless of what it actually is.
+ *
+ * The requirements are:
+ *
+ * - must not be the root controller, any of the following:
+ *   - it supports legend
+ *   - it supports having the layer opacity being changed
+ *
+ * @return {boolean} Whether the layer tree controller supports being
+ *     "customized" or not.
+ * @export
+ */
+ngeo.LayertreeController.prototype.supportsCustomization = function() {
+  return !this.isRoot &&
+    (
+      this.supportsLegend() ||
+      this.supportsOpacityChange()
+    );
+};
+
+
+/**
+ * @return {boolean} Whether the layer tree controller supports having a
+ *     legend being shown.
+ * @export
+ */
+ngeo.LayertreeController.prototype.supportsLegend = function() {
+  return this.node.metadata && this.node.metadata.legend;
+};
+
+
+/**
+ * @return {boolean} Whether the layer tree controller supports having its
+ *     layer opacity being changed or not.
+ * @export
+ */
+ngeo.LayertreeController.prototype.supportsOpacityChange = function() {
+  return this.layer &&
+    (
+      (
+        this.depth === 1 && !this.node['mixed']
```

Comment:
Instead of using the quoted notation, can you try to change `this.node` type from `Object` to `gmfThemes.GmfGroup` ?

Reply:
+1

Reply:
with an assert

Reply:
We can't.  This is a file from "ngeo".  We can't put code from "gmf" there.

Reply:
@fredj @sbrunner let me know if this is okay.

Reply:
@sbrunner can we do an exception for the externs ?

Reply:
instead of changing the `@type` here, these two function must be moved from `ngeo` to `gmf`